mesonbuild/backend/ninjabackend.py

```diff
@@ -3417,22 +3417,28 @@ def generate_utils(self):
         self.add_build(elem)
 
     def generate_ending(self):
-        for targ, deps in [
-                ('all', self.get_build_by_default_targets()),
-                ('meson-test-prereq', self.get_testlike_targets()),
-                ('meson-benchmark-prereq', self.get_testlike_targets(True))]:
-            targetlist = []
-            # These must also be built by default.
-            # XXX: Sometime in the future these should be built only before running tests.
-            if targ == 'all':
-                targetlist.extend(['meson-test-prereq', 'meson-benchmark-prereq'])
-            for t in deps.values():
-                # Add the first output of each target to the 'all' target so that
-                # they are all built
-                targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0]))
-
-            elem = NinjaBuildElement(self.all_outputs, targ, 'phony', targetlist)
-            self.add_build(elem)
+        default_targetlist = []
+        for t in self.get_build_by_default_targets().values():
+            # Add the first output of each target to the 'all' target so that
+            # they are all built
+            default_targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0]))
+
+        default_elem = NinjaBuildElement(self.all_outputs, 'all', 'phony', default_targetlist)
+        self.add_build(default_elem)
+
+        test_targetlist = []
+        for t in self.get_testlike_targets(benchmark=False).values():
+            test_targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0]))
+
+        test_elem = NinjaBuildElement(self.all_outputs, 'meson-build-tests', 'phony', test_targetlist)
```

Comment:
Why rename meson-test-prereq and meson-enchmark-prereq? Isn't the relevant thing to change here just whether to do the 
```python
            if targ == 'all':
                targetlist.extend(['meson-test-prereq', 'meson-benchmark-prereq'])
```
piece?

Reply:
Good call! I've taken a deeper dive on the fix, and tried to stay as close as I could to the paradigms of the current code